json-schema/json-schema-generic/src/test/scala/endpoints4s/generic/JsonSchemasCirceTest.scala

```diff
@@ -0,0 +1,32 @@
+package endpoints4s
+package generic
+
+import io.circe.Json
+import org.scalatest.freespec.AnyFreeSpec
+
+class JsonSchemasCirceTest extends AnyFreeSpec {
```

Comment:
Do the other (existing) tests cross-compile? What is the purpose of this new test?

Reply:
> Do the other (existing) tests cross-compile?

I'm not sure what you mean. AFAIK all tests in this module compiles and pass on both scala 2 and 3

> What is the purpose of this new test?

I wanted to make sure the actual encoding/decoding was working especially given all the `asInstanceOf` being done and to make sure that I didn't get an "off by one" kind of error in the change I did regarding the ordinal in `summonTagged`.

While `JsonSchemasTest` tests the generic derivation of the codecs. AFAIU, it doesn't really encode/decode anything so all the code inside the `xmap` was not tested (as shown by the coverage report)

Reply:
Yeah, I didn’t remember which tests were cross-compiled and cross-ran.

OK, thank you for expanding the test coverage. I don’t have much time to improve the test suite for now, feel free to add more tests if needed!